style_guide/README.md

```diff
@@ -0,0 +1,200 @@
+
+To contribute to a branch on another person's personal fork, add a new `remote`
+that points to their fork, and use the nominal workflow for contributing:
+
+    git remote add <username> <remote url>
+    git fetch <username>
+    git checkout -b <branchname> <username>/<branchname>
+    # Make some changes via add/commit cycle
+    git push <username> <branchname>
+
+Python Coding
+-------------
+
+`imap_processing` code shall adhere to the [PEP8](https://www.python.org/dev/peps/pep-0008/)
+conventions save for the following exceptions:
+
+ - Lines of code need not be restricted to 79 characters.  However, it is
+encouraged to break up excessively long lines into several lines if it benefits
+the overall readability of the code
+
+ Additionally, the code shall adhere to the following special guidelines:
+
+ - Function and class definitions should be placed in alphabetical order in the
+module
+ - It is encouraged to annotate variables and functions using the
+[`typing`](https://docs.python.org/3/library/typing.html) library (see
+[PEP 483](https://www.python.org/dev/peps/pep-0483/),
+[PEP 484](https://www.python.org/dev/peps/pep-0484/), and
+[PEP 526](https://www.python.org/dev/peps/pep-0526/)).
+
+
+Documentation
+-------------
+
+`imap_processing` code shall adhere to the [PEP257](https://www.python.org/dev/peps/pep-0008/)
+and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) conventions.
+
+The following are further recommendations:
+
+- Each module should have at minimum a description, `Authors`, and `Use`
+sections.
+- Each function/method should have at minimum a description, `Parameters` (if
+necessary), and `Returns` (if necessary) sections.
+
+Versioning
+----------
+
+Any changes pushed to the `main` branch should be tagged with a version number.
+The version number convention is `x.y.z`, where
+
+    x = The main version number.  Increase when making incompatible API changes.
+    y = The feature number.  Increase when change contains a new feature with or without bug fixes.
+    z = The hotfix number. Increase when change only contains bug fixes.
+
+
+Security
+--------
+
+The following items should never be committed in the `imap_processing` source
+code or GitHub issues/pull requests:
+
+- Account credentials of any kind (e.g. database usernames/passwords, AWS
+credentials, etc.)
+- Internal directory structures or filepaths
+- Machine names
+- Proprietary data
+
+If `imap_processing` code needs access to this information, it should be stored
+in a configuration file that is not part of the repository.
+
+Additionally, developers of this project should be mindful of application
+security risks, and should adhere to the
+[OWASP Top 10](https://www.owasp.org/images/7/72/OWASP_Top_10-2017_%28en%29.pdf.pdf)
+as much as possible.
+
+Naming Conventions
+------------------
+
+TBD
+
+Tools and Library Recommendations
+---------------------------------
+
+- `spicepy` for using SPICE kernels
```

Comment:
```suggestion
- `spiceypy` for using SPICE kernels
```

Maybe we should add these to the documentation build with subsections for the optional/required dependencies and links out to them once that is available?